pkg/domain/infra/abi/containers.go

```diff
@@ -1097,6 +1107,7 @@ func (ic *ContainerEngine) ContainerRun(ctx context.Context, opts entities.Conta
 		if err := ctr.Start(ctx, true); err != nil {
 			// This means the command did not exist
 			report.ExitCode = define.ExitCode(err)
+			_ = removeContainer(ctr, true)
```

Comment:
this removes the container even when opts.Rm is not set
also we should not ignore the error

Reply:
Fixed.